src/android/aidl/birthday-service/birthday_service/src/client.rs

```diff
@@ -36,7 +39,47 @@ fn main() -> Result<(), binder::Status> {
 
     binder::ProcessState::start_thread_pool();
     let service = connect().expect("Failed to connect to BirthdayService");
+
+    // Call the simple API:
     let msg = service.wishHappyBirthday(&name, years)?;
     println!("{msg}");
+
+    // ANCHOR: wish_with_provider
+    // Perform the same operation by sending an `InfoProvider` object as a `dyn
+    // IBirthdayInfoProvider`.
+    let provider = Strong::new(Box::new(InfoProvider { name, age: years as u8 }));
```

Comment:
```suggestion
    let provider = BnBirthdayInfoProvider::new_binder(InfoProvider { name, age: years as u8 }, BinderFeatures::default());
```
or something like that.